corva/models/base.py

```diff
@@ -1,27 +1,40 @@
+from typing import Any, List, Optional, TypeVar
+
 from pydantic import BaseModel, Extra
 
-from corva.event import Event
+from corva.network.api import Api
+from corva.state.redis_state import RedisState
 
 
-class BaseContext(BaseModel):
-    """Used to pass different parameter sets to steps predefined in BaseApp.run function.
+class BaseEvent:
+    pass
 
-    Child classes of BaseApp may need:
-      1 unique sets of parameters passed to each step (e.g.
-        TaskApp.process(event, task_data) vs StreamApp.process(event, state))
-      2 save data in some step, that will be used in the other one
 
-    Instead of bloating BaseApp's steps with obsolete parameters (e.g. BaseApp.process(event, task_data, state),
-      see above that `task_data` in used only in TaskApp and `state` - in StreamApp), context instances are used
-      to contain all necessary parameters for app to run.
-    """
+class BaseContext(BaseModel):
+    """Stores common data for running a Corva app."""
 
     class Config:
         arbitrary_types_allowed = True
+        extra = Extra.allow
+
+    raw_event: str
+    app_key: str
 
-    event: Event
+    event: Optional[BaseEvent] = None
+    api: Optional[Api] = None
+    state: Optional[RedisState] = None
+    user_result: Any = None
 
 
 class BaseEventData(BaseModel):
     class Config:
         extra = Extra.allow
+
+
+BaseEventDataTV = TypeVar('BaseEventDataTV', bound=BaseEventData)
```

Comment:
Do we need this anymore now that this isn't nested?

Reply:
it is needed for correct type hints in ListEvent instances. ListEvent class was deleted in pull 14 at all